artifacts/scripts/util.sh

```diff
@@ -214,12 +214,12 @@ sync_repo() {
                 echo "Couldn't find a ${commit_msg_tag} commit SHA in any commit on ${dst_branch}."
                 return 1
             fi
-            local k_base_merge=$(git-find-merge ${k_base_commit} upstream/${src_branch})
-            if [ -z "${k_base_merge}" ]; then
-                echo "Didn't find merge commit of source commit ${k_base_commit}. Odd."
-                return 1
-            fi
-            git branch -f filtered-branch-base ${k_base_merge} >/dev/null
+            #local k_base_merge=$(git-find-merge ${k_base_commit} upstream/${src_branch})
+            #if [ -z "${k_base_merge}" ]; then
+            #    echo "Didn't find merge commit of source commit ${k_base_commit}. Odd."
+            #    return 1
+            #fi
+            git branch -f filtered-branch-base ${k_base_commit} >/dev/null
 
```

Comment:
at this position we can check with `git diff --exit-code --quiet ${filtered-branch-base} ${filtered-branch} -- "${subdirectory}"` whether there was any change. If there hasn't been one, we can return and safe the runtime of the filter-branch rewrite. This will easily cut down the runtime by 30sec per branch.

Reply:
not sure I understand... you want to replace git branch -f with git diff?

Reply:
https://github.com/kubernetes/publishing-bot/pull/41